### ISSUE TITLE
BUG REPORT: Since update to 3.1.0 Aerial doesn't save preferences reverting back everything!

### ISSUE THREAD

**Issue by LoopNCB** [JohnCoates/Aerial#1262]
Hi there! 

- [ ] Mac model: MacBook Pro 2016 4xTBP3 - Intel i5 2,9Ghz - 512 SSD
- [ ] macOS version: MacOS Monterey 12.5.1
- [ ] Monitor setup: Integrated Retina Display

>> Debug File: 
[AerialLog copia.txt](https://github.com/JohnCoates/Aerial/files/9420458/AerialLog.copia.txt)

#### Description of issue:

So Aerial was working wonderfully on my actual and current setup until 3.1.0 😔 that I updated with the aerial companion app 
(version 1.2.2). Also wanted to mention that I currently have only 30 GB left of storage on SSD... FYI. 

I uninstalled the aerial companion app and reinstalled it, but the same bug goes on and on... so what happens exactly? 
When I set up everything the way I want (a simple personalization where I only use 5 fix aerial city videos  -videos that I save on my machine- plus deactivating any kind of automatic and new downloads) the setup menu, does not save my choices after all and reverts to its initial state when going back to the menu or launching the screensaver. Also worth to mention that the "welcome splash screen" comes up over and over again when I go into the settings panel to personalize my screensaver settings.

I obviously would like to revert to 3.0.9 -found the file here- but first I'll wait for some feedback as it might be something else since I'm really not a PRO in these fields... who knows what I might be missing... therefore I attached the debug log file so you can have a look at it.

I just LOVE your screensaver and really appreciate your huge effort. 🔝 🔝 🔝  

Hopefully you can help me out with this issue! 😃

**Comment by glouel** [comment]
Hey @LoopNCB 

That's super not good. I'll try to have a look at your file tonight and let you know. By any chance, did you move your user folder to some other place ? I did try to improve settings but may have messed something for some people.

If there's anything peculiar about your machine (some sort of corporate management, stuff moved to external drives, etc etc), let me know, it may help me narrow it down !

Sorry for the issue again :(

**Comment by glouel** [comment]
@LoopNCB on top of what's said above, can you try opening again the settings, with debug mode enabled, and send me the log afterwards ? I'm missing the interesting bits (the debug startup), but if you can't save settings at all that might be an issue.

**Comment by glouel** [comment]
It's not so much you doing it wrong that some mess I made. I'm not sure why still. If I make a quick build, would you mind installing it and testing it quickly ? I'll force enable the logs and it should give me a better sense of what's happenning. 

(short term fix in any case, revert to 3.0.9)

**Comment by glouel** [comment]
@LoopNCB if you can,

- Delete your current AerialLog.txt

- Then please install  this build : 

https://github.com/glouel/Aerial/releases/download/v3.1.1test1/Aerial.saver.zip

- After installing try opening settings, then grab the AerialLog.txt, this one should contain what I need. Thanks for your help !

**Comment by glouel** [comment]
Thanks a whole lot for your patience, I think I'm getting somewhere. What seems to happen is that the new preferences file isn't found, so you always revert to a brand new configuration. But some details don't fully make sense yet.

If you are able to, can you open "Terminal" on your mac and type : 
```
cd ~/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/
```
then, assuming there was no error, type : 
```
ls -la
```
and paste the output

Optional step : 

```
cd ByHost
```
then
```
ls -la
```
and paste the output too.

I'll try to think of a way to workaround whatever seems not to be working, but it's a bit puzzling yet ! Thanks for your help !

**Comment by dankudos** [comment]
Same issue here though my home folder is not on my boot Volume. Very strong agree that the application is not setting/finding a preference file so I'm just getting a fresh experience on every launch. 

Not to cross streams but here's the output you asked for though it's not much at all.

> Media-Mac-3:~ media$ cd ~/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/
Media-Mac-3:Preferences media$ ls -la
total 48
drwx------   6 media  admin   204 Aug 21 15:03 .
drwx------  31 media  admin  1054 Jul 30 21:54 ..
drwx------@  3 media  staff   102 Aug 26 11:01 ByHost
-rw-------   1 media  staff   197 Aug 21 15:03 com.apple.ScreenSaver.Engine.legacyScreenSaver.plist
lrwxr-xr-x   1 media  admin    51 Nov 11  2020 com.apple.security.plist -> ../../../../../Preferences/com.apple.security.plist
lrwxr-xr-x   1 media  admin    58 Nov 11  2020 com.apple.security_common.plist -> ../../../../../Preferences/com.apple.security_common.plist
Media-Mac-3:Preferences media$ cd ./ByHost
Media-Mac-3:ByHost media$ ls -la
total 16
drwx------@ 3 media  staff   102 Aug 26 11:01 .
drwx------  6 media  admin   204 Aug 21 15:03 ..
-rw-r--r--@ 1 media  staff  6148 Aug 26 11:01 .DS_Store
Media-Mac-3:ByHost media$

**Comment by glouel** [comment]
@dankudos thanks a whole lot for the info, I believe I may have fixed the issue for home folder not on boot volume for upcoming version.

HOWEVER

Your log output is seriously concerning, this is where your "old" and new aerial settings should be, yet they aren't. There's something I'm not understanding here, and I think it may be some x86/arm container mess biting back. 

@dankudos can you try this for me ? 
- open Aerial settings, Advanced, Show log in finder
- in finder right click/ctrl click AerialLog.txt to show the contextual menu, press alt to see the alternate options, and 2/3 down the list, press "Copy AerialLog.txt" as path and give me that path ?
- from that Finder window, go two level down, then one level up to Preferences/ then ByHost/ 
- what's in that folder ?

**Comment by dankudos** [comment]
I was just getting read to double post that I've provided you bad info due to my own troubleshooting. I manually removed the old pref file earlier to confirm it wasn't "stuck" or otherwise the issue. As you see above no new file was generated. This is why you see the modified date on the ByHost folder as today. I've moved that file back and provided similar output below. You can see this pref file has not been modified since the 3.1 update on 12 August.

> Media-Mac-3:ByHost media$ cd ~/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/
Media-Mac-3:Preferences media$ ls -la
total 48
drwx------   6 media  admin   204 Aug 21 15:03 .
drwx------  32 media  admin  1088 Aug 26 11:58 ..
drwx------@  4 media  staff   136 Aug 26 12:02 ByHost
-rw-------   1 media  staff   197 Aug 21 15:03 com.apple.ScreenSaver.Engine.legacyScreenSaver.plist
lrwxr-xr-x   1 media  admin    51 Nov 11  2020 com.apple.security.plist -> ../../../../../Preferences/com.apple.security.plist
lrwxr-xr-x   1 media  admin    58 Nov 11  2020 com.apple.security_common.plist -> ../../../../../Preferences/com.apple.security_common.plist
Media-Mac-3:Preferences media$ cd ./ByHost
Media-Mac-3:ByHost media$ ls -la
total 32
drwx------@ 4 media  staff   136 Aug 26 12:02 .
drwx------  6 media  admin   204 Aug 21 15:03 ..
-rw-r--r--@ 1 media  staff  6148 Aug 26 12:02 .DS_Store
-rw-------  1 media  staff  6308 Aug 12 11:25 com.JohnCoates.Aerial.A3176DE3-6086-5767-B71F-21EDD8946713.plist
Media-Mac-3:ByHost media$

**Comment by glouel** [comment]
Ha I see, well this is less concerning then ! 

I'm not 100% sure I fully understand the issue yet, I tried to do things more properly and hopefully it will be ok this time. 

I'll have a test build ready in a few minutes if you feel like testing it. Thanks for your help in any case !

**Comment by glouel** [comment]
@dankudos if you can please try installing this : 
https://github.com/glouel/Aerial/releases/download/v3.1.1test2/Aerial.saver.zip

Hopefully it will work as is. If not, I'm interested if there's an aerial file in Preferences (not byHost) created this time or not.

**Comment by dankudos** [comment]
Seems to be working as expected this time. It did not launch the Welcome setup dialog and read in my existing prefs as well as taking new settings. Also, there's your name on a pref in the desired directory. Grand success!

> Media-Mac-3:ByHost media$ cd ~/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/
Media-Mac-3:Preferences media$ ls -la
total 80
drwx------   7 media  admin   238 Aug 26 12:55 .
drwx------  32 media  admin  1088 Aug 26 11:58 ..
drwx------@  4 media  staff   136 Aug 26 12:02 ByHost
-rw-------   1 media  staff   197 Aug 21 15:03 com.apple.ScreenSaver.Engine.legacyScreenSaver.plist
lrwxr-xr-x   1 media  admin    51 Nov 11  2020 com.apple.security.plist -> ../../../../../Preferences/com.apple.security.plist
lrwxr-xr-x   1 media  admin    58 Nov 11  2020 com.apple.security_common.plist -> ../../../../../Preferences/com.apple.security_common.plist
-rw-------   1 media  staff  8955 Aug 26 12:55 com.glouel.Aerial.plist
Media-Mac-3:Preferences media$

**Comment by glouel** [comment]
🥳

Hopefully @LoopNCB will confirm this build also fix the original issue. 

I had to move the settings from the ByHost directory to one level down in order to be able to share settings with Aerial Companion (Companion couldn't use the ByHost file, sadly). I thought I had it correct, but I made a huge mess 😅

From now on settings will be using this shared file. 

Waiting for confirmation that the original issue is fixed and I'll release 3.1.1. Thanks for taking the time to test the new build !

**Comment by glouel** [comment]
Thanks @LoopNCB for the confirmation that it works now, and no worries, I do appreciate greatly when people take the time to report issues.  

Thanks both for your help fixing this mess, I just released 3.1.1 with the fix (it's identical to test2, except logs are no longer enabled by default), so you can resume using/updating via Companion. 

Take care and thanks for your kind words !
